oak_channel/SPEC.md

```diff
@@ -1,39 +1,38 @@
-# Oak Baremetal Communication Protocol Specification
+# Oak Restricted Kernel Communication Protocol Specification
 
 This document serves as a specification of the protocol used for communication
-between the Untrusted Launcher and Trusted Runtime. The Trusted Runtime should
-be able to communicate with any implementation of the Untrusted Launcher that
-implements this specification.
+between the Untrusted Launcher (running on the host) and Enclave Binry built
+with the Restricted Kernel (running inside a guest VM).
 
-In the specification below the Trusted Runtime will henceforth be referred to
-the as the "service". The Untrusted Launcher will be referred to the as the
-"client".
+In the specification below the Enclave Binary will be referred to the as the
+"service". The Untrusted Launcher will be referred to the as the "client".
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
 "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
 interpreted as described in [RFC 2119](https://www.rfc-editor.org/rfc/rfc2119).
 
 ## Layer Defintions
 
-Conceptually, the Oak Baremetal Communication Protocol is made up of three
-layers. The [Method Layer](#method-layer) uses the Oak IDL to define the methods
-exposed by the service. It is responsible for encoding method parameters and
-return values. The [Message](#message-layer) takes encoded parameters / return
-values and wraps them as distinct messages that include metadata. The
-[Frame](#frame-layer) creates individual frames from messages, which are then
-sent over the transport channel.
+Conceptually, the Oak Restricted Kernel Communication Protocol is made up of
+three layers. The [Invocation Layer](#invocation-layer) uses protocol buffers
```

Comment:
It is not clear to me what is the lowest layer of the stack here. From the description I expect the message layer to be at the highest level of the stack but the fact that is stated in the middle makes me uncertain about that.

Reply:
I found this a bit hard to read as well; consider making it a bulleted list.